File: java/JavaTupleEngine/src/it/mintlab/desktopjava/tupleengine/TupleEngine.java
package it.mintlab.desktopjava.tupleengine;

	public final class TupleEngine
	{
		public static Term parse(String tuple)
		{
			return new Parser().getTerms(new Lexer().getTokens(tuple));
		}

		public static Term parse(String tuple, Object[] parameters)
		{
			//java.util.HashMap<String, AtomToken> parTokens = new java.util.HashMap<String,AtomToken>();
			java.util.ArrayList<IToken> tokens = new Lexer().getTokens(tuple);
			for (IToken token : tokens)
			{
				int outvalue = 0;
				String tokenName = token.toString();
				if (token instanceof AtomToken && tokenName.startsWith("$")){
					try{
						outvalue = Integer.parseInt(tokenName.substring(1));
						((AtomToken)token).setParameter(parameters[outvalue]);
					}catch(NumberFormatException nfe){}
				}
			}
			return new Parser().getTerms(tokens);
		}


	}

File: compact-net/CompactNetTupleEngine/TupleEngine.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace it.mintlab.mobilenet.tupleengine
{
    public static class TupleEngine
    {
        public static Term parse(string tuple)
        {
            return new Parser().getTerms(new Lexer().getTokens(tuple));
        }

        public static Term parse(string tuple, object[] parameters)
        {
            List<IToken> tokens = new Lexer().getTokens(tuple);
            foreach (IToken token in tokens)
            {
                string tokenName = token.ToString();
                if (token is AtomToken && tokenName.StartsWith("$") && tokenName.Length >= 2)
                {
                    try
                    {
                        int i = int.Parse(tokenName.Substring(1));
                        ((AtomToken)token).setParameter(parameters[i]);
                    }
                    catch (Exception ex)
                    {}
                }
            }
            return new Parser().getTerms(tokens);
        }



    }
}
